### PR TITLE
monasca: Add tempest tests

### PR DESCRIPTION
**Pull request by nicolasbock**
Add tempest tests for monasca.

### CODE REVIEW

**Comment by nicolasbock** [comment]
I pushed this up here to be able to test against Jenkins.

**Comment by nicolasbock** [comment]
Failing on ceilometer right now :frowning_face:

**Comment by nicolasbock** [comment]
I have tested this PR combined with https://github.com/crowbar/crowbar-openstack/pull/942 on `mkchp` and it is running smoke and monasca tests.

File: scripts/mkcloud [hunk 1]
```diff
@@ -757,9 +757,8 @@ function testtype
     # defcore test.  A full test is assumed by default.
     local type="full"
     if [[ $tempestoptions ]] ; then
-        if [[ $tempestoptions =~ -s ]] ; then
+        if [[ $tempestoptions =~ -s || $tempestoptions =~ smoke ]] ; then
```
Comment (dirkmueller) [inline scripts/mkcloud]: unrelated fix, separate commit?

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -4095,6 +4095,12 @@ function oncontroller_testsetup
         fi
     fi
 
+    if [[ $want_monasca_proposal = 1 ]]; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: I don't like such magic rewritings.. would it be okay to just leave that part out? I think it doesn't serve a lot of purpose.. the smoke tests should still be runnable (they just won't test monasca then).

**Comment by nicolasbock** [comment]
@dirkmueller updated the PR.

**Comment by nicolasbock** [comment]
Some of the monasca tempest tests are failing. I am going to fix before we should merge this PR.

**Comment by nicolasbock** [comment]
(Otherwise we will break CI right away)

**Comment by nicolasbock** [comment]
On the other hand, maybe we should merge it so that CI is broken :smile:

File: jenkins/ci.suse.de/templates/cloud-mkcloud-job-monasca-template.yaml [hunk 3]
```diff
@@ -23,6 +23,7 @@
             nodenumbercompute=1
             want_monasca_proposal=1
             storage_method=none
+            tempestoptions=--serial --regex '^monasca_tempest_tests.tests.api'
```
Comment (vmoravec) [inline jenkins/ci.suse.de/templates/cloud-mkcloud-job-monasca-template.yaml]: There are also 13 tests in `monasca_log_api_tempest.tests.log_api` namespace, is there any reason to avoid coverage there?

Comment (nicolasbock) [inline jenkins/ci.suse.de/templates/cloud-mkcloud-job-monasca-template.yaml]: No, not really. I just missed them :smile:

Comment (nicolasbock) [inline jenkins/ci.suse.de/templates/cloud-mkcloud-job-monasca-template.yaml]: @vmoravec Fixed. Please have a look now.

File: scripts/mkcloud (hunk 1 above)
Comment (nicolasbock) [inline scripts/mkcloud]: Done.

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (nicolasbock) [inline scripts/qa_crowbarsetup.sh]: Removed.

**Comment by nicolasbock** [comment]
@dirkmueller Could you give it another look?

**Comment by toabctl** [comment]
retriggered CI (including a monasca job)

**Comment by jdsn** [comment]
@nicolasbock is this still wip? if not please remove the label so we can merge this PR soon, thanks.

**Comment by nicolasbock** [comment]
@jdsn No, I removed the label.

**Comment by jdsn** [comment]
@nicolasbock Is this PR still valid? Retriggering gating jobs.

**Comment by nicolasbock** [comment]
It's been sitting here for a long time @jdsn . It's probably still valid, but I will need to look into it again.

**Comment by jdsn** [comment]
@nicolasbock please merge after you had your "needed look" :)
